dll/ntdll/ldr/ldrutils.c

```diff
@@ -2249,11 +2249,12 @@ LdrpCheckForLoadedDll(IN PWSTR DllPath,
 
 NTSTATUS
 NTAPI
-LdrpGetProcedureAddress(IN PVOID BaseAddress,
-                        IN PANSI_STRING Name,
-                        IN ULONG Ordinal,
-                        OUT PVOID *ProcedureAddress,
-                        IN BOOLEAN ExecuteInit)
+LdrpGetProcedureAddress(
+    _In_ PVOID BaseAddress,
+    _In_opt_ PANSI_STRING Name,
+    _In_ ULONG Ordinal,
```

Comment:
It looks like `Ordinal` is optional too: only one of them is used for a given execution.

Reply:
> It looks like `Ordinal` is optional too: only one of them is used for a given execution.

Yes, committed. I use `_When_` to indicate either `Name` or `Ordinal`. Tested:

```C
NTSTATUS
NTAPI
LdrGetProcedureAddress(
    _In_ PVOID BaseAddress,
    _In_opt_ _When_(Ordinal == 0, _Notnull_) PANSI_STRING Name,
    _In_opt_ _When_(Name == NULL, _In_range_(>, 0)) ULONG Ordinal,
    _Out_ PVOID *ProcedureAddress
);

// No warning
Status = LdrGetProcedureAddress(hKernel32,
                                &(ANSI_STRING)RTL_CONSTANT_STRING("AcquireSRWLockExclusive"),
                                0,
                                &p1);

// No warning
Status = LdrGetProcedureAddress(hKernel32, NULL, 1, &p2);

// Warning C28020: The expression '_Param_(3)>0' is not true at this call.
// Warning C6387: '_Param_(2)' could be '0'.
Status = LdrGetProcedureAddress(hKernel32, NULL, 0, &p2);
```